texastribune/salesforce-stripe
# Add security and rate limiting features

#### What's this PR do?

Adds several security-related headers and rate-limiting. Adds a Content Security Policy that's "report only" at the moment so we can confirm everything is working as expected before we start enforcing it.

#### Why are we doing this? How does it help us?

It's good to be more secure. Especially when handling credit cards. 

#### How should this be manually tested?

1. set `REPORT_URI` in your `env-docker` to the value in Sentry (or ask me.) Start the server.
1. Visit a few pages. Confirm everything's working as intended. 
1. Comment out line 62 of `app.py`. Restart the server. Confirm that you get a warning on the browser console and an alert in Sentry.

#### Have automated tests been added?

No.

#### Has this been tested on mobile?

No.

#### Are there performance implications?

Not likely. 

#### What are the relevant tickets?

https://3.basecamp.com/3098728/buckets/736178/todos/1322629724

#### How should this change be communicated to end users?

N/A

#### Next steps?

See if we can remove `data:`, `unsafe-inline`, and `unsafe-eval`.
